PG: Updates to config

jamessewell opened the pull request:
- added some values in postgresql.conf
- added support for sync replication
- changed creation of REPLICATION user to be async to avoid deadlock
- added an option to use data checksums for initdb

jamessewell committed:
Updates to config:

- added some values in postgresql.conf
- added support for sync replication
- changed creation of REPLICATION user to be async to avoid deadlock
- added an option to use data checksums for initdb

bodymindarts commented on postgresql/default.toml:
```diff
@@ -11,6 +12,8 @@ password = 'admin'
 [replication]
 name = 'replication'
 password = 'replication'
+synchronous = true
```
Don't think this should be on by default

jamessewell replied:
Agree

bodymindarts commented on postgresql/default.toml:
```diff
@@ -11,6 +12,8 @@ password = 'admin'
 name = 'replication'
 password = 'replication'
+synchronous = true
+synchronous_standby_names = '*'
```
Don't know if this should be editable. Ideally hab would just figure everything out by itself.

jamessewell replied:
I think we have a lot of customers who would want this - but to make it useful you actually need to manage the `application_name` of the standby servers. Probably too complex for the initial release. 

I'll take it out and always use * and maybe return to it later.

bodymindarts commented on postgresql/hooks/init:
```diff
@@ -20,7 +20,7 @@ if [[ ! -f "{{pkg.svc_data_path}}/PG_VERSION" ]]; then
     -D {{pkg.svc_data_path}} \
     --pwfile {{pkg.svc_config_path}}/pwfile \
-    --locale POSIX
+    --locale POSIX {{ #if cfg.data_checksums }}--data-checksums{{ /if }}
```
From reading a bit about this setting I would say we have it on by default or? It protects against data corruption but doesn't seem to have any downside to it.

bodymindarts commented on postgresql/default.toml:
```diff
@@ -1,7 +1,8 @@
-log_line_prefix = "%m: "
+log_line_prefix = '%t [%p]: [%l-1] user=%u,db=%d,client=%h %r (%x:%e)'
 log_level = 'ERROR'
+data_checksums = true
```
Why would anyone wan't to turn this off? Does it really need to be configurable?

bodymindarts replied:
Okay so it makes things a bit slower.... so it allows ppl to choose fast over correct I guess.

jamessewell replied:
Good point actually, I think you are right and we just leave it on till there is a use case for off?

jamessewell committed:
Fixed up from review:

- synchronous_standby_names = '*' by default when synchronous mode is enabled
- synchronous mode disabled by default
- data_checksums on by default with no config option

Commit:
Merge branch 'master' into config_updates
